### ISSUE TITLE
Module and component generators are slow

### ISSUE THREAD

**Issue by sebawita** [NativeScript/nativescript-schematics#185]
**Environment**
`@nativescript/schematics 0.4 and 0.5`

**Describe the bug**
Running all of the following commands in a code-sharing project takes too long to execute:

`ng g m my-module`
`ng g c my-component --no-web`
`ng g migrate-module web-component`

**To Reproduce**

```
# Create a new code-sharing project with Angular CLI and NativeScript Schematics
ng new my-app -c=@nativescript/schematics --shared`
cd my-app

# Test - generate component
ng g c my-component --no-web

# Test - generate module
ng g m my-module

# Test - migrate-module
ng g m web-module --no-nativescript
ng g migrate-module --name=web-module
```

**Expected behavior**
These commands should execute in under 2 seconds.

**Sample project**
n/a

**Additional context**
It looks like `filter` from **angular schematics** is the function that slows the whole process a lot, as it goes through all `node_modules` files **one by one**.

```
import { filter } from '@angular-devkit/schematics'
```

**Comment by erkanarslan** [comment]
In my case `ng g module` takes around 30 seconds but `ng g component` is executed in a few seconds.

**Comment by sebawita** [comment]
In order to fix this, we need to resolve #189 first, as the faulty tests block me from pushing a fix.

**Comment by sis0k0** [comment]
The problem is still present after the fix was released in `0.5.2`. I reproduced it in the [web-mobile-project](https://github.com/nativescript/web-mobile-project).
![component-generation](https://user-images.githubusercontent.com/7893485/55310007-18263580-5468-11e9-8497-9adb4dde9a05.png)
